packages/firestore/lite/src/api/reference.ts

```diff
@@ -350,22 +375,34 @@ export function doc<T>(
     relativePath = AutoId.newId();
   }
   validateArgType('doc', 'non-empty string', 2, relativePath);
-  const path = ResourcePath.fromString(relativePath!);
+
   if (parent instanceof Firestore) {
-    validateDocumentPath(path);
+    const absolutePath = ResourcePath.fromString(relativePath!);
+    validateDocumentPath(absolutePath);
     return new DocumentReference(
       parent,
-      new DocumentKey(path),
+      new DocumentKey(absolutePath),
       /* converter= */ null
     );
   } else {
-    const coll = cast(parent, CollectionReference);
-    const absolutePath = coll._path.child(path);
+    if (
+      !(parent instanceof DocumentReference) &&
+      !(parent instanceof CollectionReference)
+    ) {
+      throw new FirestoreError(
+        Code.INVALID_ARGUMENT,
+        'Expected first argument to collection() to be a CollectionReference, ' +
+          'a DocumentReference or FirebaseFirestore'
+      );
+    }
+    const absolutePath = ResourcePath.fromString(
+      `${parent.path}/${relativePath}`
+    );
     validateDocumentPath(absolutePath);
     return new DocumentReference(
-      coll.firestore,
+      parent.firestore,
       new DocumentKey(absolutePath),
-      coll._converter
+      parent instanceof CollectionReference ? parent._converter : null
```

Comment:
Can you add a test to make sure the converter is properly passed/omitted?

Reply:
Done